hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -110,7 +111,8 @@ private HoodieFileGroupReader(HoodieReaderContext<T> readerContext, HoodieStorag
     readerContext.setTablePath(tablePath);
     readerContext.setLatestCommitTime(latestCommitTime);
     boolean isSkipMerge = ConfigUtils.getStringWithAltKeys(props, HoodieReaderConfig.MERGE_TYPE, true).equalsIgnoreCase(HoodieReaderConfig.REALTIME_SKIP_MERGE);
-    readerContext.setShouldMergeUseRecordPosition(readerParameters.useRecordPosition() && !isSkipMerge && readerContext.getHasLogFiles());
+    boolean isParquetBaseFile = inputSplit.getBaseFileOption().map(HoodieBaseFile::toString).map(path -> path.endsWith(HoodieFileFormat.PARQUET.getFileExtension())).orElse(false);
```

Comment:
we can add a flag inside the InputSplit to holds the variable `isParquetBaseFile`

Reply:
Updated